core/src/test/java/io/confluent/kafka/schemaregistry/SASLClusterTestHarness.java

```diff
@@ -0,0 +1,182 @@
+    kdc.start();
+
+    krb5ConfPath = System.getProperty("java.security.krb5.conf");
+
+    createPrincipal(zkServerKeytab, "zookeeper/localhost");
+    createPrincipal(kafkaKeytab, "kafka/localhost");
+    createPrincipal(schemaRegistryKeytab, "schema-registry/localhost");
+
+    // destroy SASL settings so all tests don't have SASL enabled. SASL is re-enabled in a @Before method below.
+    destroySaslHelper();
+  }
+
+  @Before
+  @Override
+  public void setUp() throws Exception {
+    destroySaslHelper();
+    System.setProperty(JAAS_CONF, jaasFile.getAbsolutePath());
+    System.setProperty(KRB5_CONF, krb5ConfPath);
+    System.setProperty(ZK_AUTH_PROVIDER, "org.apache.zookeeper.server.auth.SASLAuthenticationProvider");
+    super.setUp();
+  }
+
+  private static void createPrincipal(File keytab, String principalNoRealm) throws Exception {
+    Seq<String> principals = scala.collection.JavaConversions.asScalaBuffer(
+            Arrays.asList(principalNoRealm)
+    ).seq();
+    kdc.createPrincipal(keytab, principals);
+  }
+
+  private static JaasTestUtils.JaasSection createJaasSection(File keytab, String principalWithRealm,
+                                                      String jaasContextName, String serviceName) {
+    final scala.Option<String> serviceNameOption = scala.Option.apply(serviceName);
+    JaasTestUtils.Krb5LoginModule krbModule = new JaasTestUtils.Krb5LoginModule(true, true,
+            keytab.getAbsolutePath(), principalWithRealm, true, serviceNameOption);
+    Seq<JaasTestUtils.JaasModule> jaasModules = scala.collection.JavaConversions.asScalaBuffer(
+            Arrays.asList(krbModule.toJaasModule())
+    ).seq();
+    return new JaasTestUtils.JaasSection(jaasContextName, jaasModules);
+  }
+
+  @Override
+  protected KafkaConfig getKafkaConfig(int brokerId) {
+    final Option<File> trustStoreFileOption = scala.Option.apply(null);
+    final Option<SecurityProtocol> saslInterBrokerSecurityProtocol =
+            scala.Option.apply(SecurityProtocol.SASL_PLAINTEXT);
+    Properties props = TestUtils.createBrokerConfig(
+            brokerId, zkConnect, false, false, TestUtils.RandomPort(), saslInterBrokerSecurityProtocol,
+            trustStoreFileOption, EMPTY_SASL_PROPERTIES, false, true, TestUtils.RandomPort(),
+            false, TestUtils.RandomPort(),
+            false, TestUtils.RandomPort(), Option.<String>empty());
+
+    injectProperties(props);
+    props.setProperty("zookeeper.connection.timeout.ms", "30000");
+    props.setProperty("sasl.mechanism.inter.broker.protocol", "GSSAPI");
+    props.setProperty(SaslConfigs.SASL_ENABLED_MECHANISMS, "GSSAPI");
+
+    return KafkaConfig.fromProps(props);
+  }
+
+  @After
+  @Override
+  public void tearDown() throws Exception {
+    destroySaslHelper();
+    super.tearDown();
+  }
+
+  @AfterClass
+  public static void tearDownKdc() throws Exception {
+    if (kdc != null) {
+      kdc.stop();
+    }
+    if (kdcHome != null && !kdcHome.delete()) {
```

Comment:
I think you need a recursive delete method right? There is one in Kafka's `Utils.delete`.